homeassistant/components/zone/__init__.py

```diff
@@ -362,10 +369,16 @@ async def async_added_to_hass(self) -> None:
         await super().async_added_to_hass()
         person_domain = "person"  # avoid circular import
         persons = self.hass.states.async_entity_ids(person_domain)
-        object_id = split_entity_id(self.entity_id)[1]
         for person in persons:
             state = self.hass.states.get(person)
-            if state and state.state == object_id:
+            if not state or None in (
+                latitude := state.attributes.get(ATTR_LATITUDE),
+                longitude := state.attributes.get(ATTR_LONGITUDE),
+                accuracy := state.attributes.get(ATTR_GPS_ACCURACY),
+            ):
+                continue
+            zone_state = async_active_zone(self.hass, latitude, longitude, accuracy)  # type: ignore[arg-type]
```

Comment:
To fix the type ignore

```suggestion
            if (
                state is None
                or (latitude := state.attributes.get(ATTR_LATITUDE)) is None
                or (longitude := state.attributes.get(ATTR_LONGITUDE)) is None
                or (accuracy := state.attributes.get(ATTR_GPS_ACCURACY)) is None
            ):
                continue
            zone_state = async_active_zone(self.hass, latitude, longitude, accuracy)
```